### ISSUE TITLE
Consider adding "requirement" to "issue"

### ISSUE THREAD

**Issue by amok** [fabian-hiller/valibot#250]
It would be nice to have the validator arguments (aka requirements, f.i see [here](https://github.com/fabian-hiller/valibot/blob/main/library/src/validations/maxLength/maxLength.ts#L13)) specified in the issue object

Consider the following example

```typescript
const Schema = object({
  password: string([minLength(8)]),
});

console.log(safeParse(Schema, { password: 'p' }).error.issues)
```

it logs out 

```js
{
	reason: "string"
	validation: "min_length"
	origin: "value"
	message: "Invalid length"
	input: "p"
	...
}
```

the ask here is to also add minLength' args to the issue object, f.i.:

```json
{
	"reason": "string"
	"validation": "min_length"
	"origin": "value"
	"message": "Invalid length"
	args: [8],
	"input": "p"
	
}
```

so they're available for the code which calls `parse`

This could be useful for error handling and i18n. For example, instead of re-generating schemas per language with `minLength(8, MESSAGE_IN_MY_LANG)`, we could do something like:

```js
try {
    const data = parse(Schema, input)
    ...
} catch (error) {
    const issue = error.issues[0]
    
    if (issue.validation === 'min_length') {
         showNotification(translate('Length should be at least {{min}}'), { min: issue.args[0] })
    }
    ...
}
```

Sometimes message' text depends on the UI context and this approach helps defining error messages closer to the place of `parse` usage.

**Comment by fabian-hiller** [comment]
Thank you for creating this issue. This is a duplicate of #191. We are in the process of improving this part of the library with #211.

**Comment by amok** [comment]
Thanks for the reply @fabian-hiller. Can't believe I missed it searching through the issues. Feel free to close this one as a duplicate.

**Comment by fabian-hiller** [comment]
That's not a problem. I am happy for everyone who contributes to Valibot, and you are now a part of it! Thank you! I close the issue once #211 is merged.

**Comment by fabian-hiller** [comment]
v0.21.0 is now available.
